src/components/Launch/LaunchWorkflowForm/CollectionInput.tsx

```diff
@@ -0,0 +1,55 @@
+import { TextField } from '@material-ui/core';
+import * as React from 'react';
+import { InputChangeHandler, InputProps, InputType } from './types';
+import { UnsupportedInput } from './UnsupportedInput';
+
+function stringChangeHandler(onChange: InputChangeHandler) {
+    return ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
+        onChange(value);
+    };
+}
+
+/** Handles rendering of the input component for a Collection of SimpleType values*/
+export const CollectionInput: React.FC<InputProps> = props => {
+    const {
+        label,
+        helperText,
+        onChange,
+        typeDefinition: { subtype },
+        value = ''
+    } = props;
+    if (!subtype) {
+        console.error(
+            'Unexpected missing subtype for collection input',
+            props.typeDefinition
+        );
+        return <UnsupportedInput {...props} />;
+    }
+    switch (subtype.type) {
+        case InputType.Blob:
+        case InputType.Boolean:
+        case InputType.Collection:
+        case InputType.Datetime:
+        case InputType.Duration:
+        case InputType.Error:
+        case InputType.Float:
+        case InputType.Integer:
+        case InputType.Map:
+        case InputType.String:
+        case InputType.Struct:
+            return (
+                <TextField
+                    helperText={helperText}
+                    fullWidth={true}
+                    label={label}
+                    multiline={true}
+                    onChange={stringChangeHandler(onChange)}
```

Comment:
Where does the input value to `stringChangeHandler` here get defined? Is it recursively referencing the `onChange` handler of the `TextField`? So, `x = stringChangeHandler(x)`?

Reply:
`stringChangeHandler` generates an event handler that will forward the value received to its argument. So in this case, the prop value ends up looking like:
```
onChange = ({ target: { value } }) => props.onChange(value); // props here are the props to CollectionInput
```